app/models/translation.rb

```diff
@@ -10,7 +13,25 @@ class Translation < ActiveRecord::Base
     scope: %i(translated_type translated_id),
     message: "translation already exists for this resource"
 
-  # TODO: Versioning and maintaining a live, published version
+  before_create :init_version
+  before_update :save_version
+
+  def init_version
+    strings.keys.each do |key|
+      string_versions[key] ||= 0
+    end
+  end
+
+  def save_version
+    if changes.include?("strings")
+      changed_keys = Translations::HashDiff.diff(*changes["strings"])
+      version = translation_versions.create!(strings: strings)
+
+      changed_keys.each do |key|
+        string_versions[key] = version.id
```

Comment:
I'm not 100% sure how this version number will be used, if it's user / UI facing then using the pk sequence number from the translations table will mean version number won't be linear for the a given resource, e.g. 1, 5, 35 etc depending how many new versions get written between updates. 

An alternative could use the length of the `translation_versions` relation instead. 

Thoughts?

Reply:
Yeah this is what I mentioned yesterday about going down the wrong path initially. This will become the version of the translated resource, set by the sync worker. The front-end will pass in the version of primary translation when it updates a foreign language translation.